include/kframework/lemmas/bitwise-simplification.k

```diff
@@ -43,4 +43,12 @@ module BITWISE-SIMPLIFICATION [symbolic]
     rule notMaxUInt160 &Int X => 0 requires #rangeUInt(160, X) [simplification]
     rule maxUInt160 &Int (X |Int (notMaxUInt160 &Int Y:Int)) => X requires #rangeUInt(160, X) andBool 0 <=Int Y [simplification]
 
+    rule [lengthBytes-upInt-32-lower-bound]:
+      lengthBytes(X) <=Int notMaxUInt5 &Int ( lengthBytes(X) +Int 31 ) => true
+      [simplification, smt-lemma]
+
+    rule [lengthBytes-upInt-32-upper-bound]:
+      lengthBytes(X) +Int 32 >Int notMaxUInt5 &Int ( lengthBytes(X) +Int 31 ) => true
+      [simplification, smt-lemma]
+
```

Comment:
Is there any reason these have to say `lengthBytes(X)`?

```suggestion
    rule [lengthBytes-upInt-32-lower-bound]:
      X <=Int notMaxUInt5 &Int ( X +Int 31 ) => true requires 0 <=Int X
      [simplification, smt-lemma]

    rule [lengthBytes-upInt-32-upper-bound]:
      X +Int 32 >Int notMaxUInt5 &Int ( X +Int 31 ) => true requires 0 <=Int X
      [simplification, smt-lemma]

```

Reply:
I don't think that work for `X >=Int pow256`, I will bound it to `[0, pow256)`.

Reply:
This breaks the backend with:
```
kore-exec: [3416348] Error (DecidePredicateUnknown):
    Failed to decide predicate:        
        /* Sfa */
        \equals{SortBool{}, _}(
            /* T Fn D Sfa Cl */ \dv{SortBool{}}("true"),
            /* T Fn D Sfa */
            Lbl'Unds'orBool'Unds'{}(
                /* T Fn D Sfa */
                Lbl'UndsEqlsEqls'Int'Unds'{}(
                    /* T Fn D Sfa */ VarUINT8:SortInt{},
                    /* T Fn D Sfa Cl */ \dv{SortInt{}}("0")
                ),
                /* T Fn D Sfa */
                Lbl'UndsEqlsEqls'Int'Unds'{}(
                    /* T Fn D Sfa */ VarUINT8:SortInt{},
                    /* T Fn D Sfa Cl */ \dv{SortInt{}}("1")
                )
            )
        )
```
on any execution. The `smt-lemma`s are not behaving well, I guess.

Reply:
Keeping `lengthBytes` for now...

Reply:
Well, was worth a try. Then I guess these lemmas shoul be moved to `include/kframework/lemmas/lemmas.k` instead, since `notMaxUInt5` is a KEVM specific constant.

Reply:
Ah, well I guess I'm just wrong then. I suppose that should be cleaned up eventually, but maybe that's a problem for another day.